openfeature/hook/__init__.py

```diff
@@ -29,6 +29,11 @@ class HookContext:
     client_metadata: typing.Optional[ClientMetadata] = None
     provider_metadata: typing.Optional[Metadata] = None
 
+    def __setattr__(self, key: str, value: typing.Any) -> None:
+        if hasattr(self, key) and key in ("flag_key", "flag_type", "default_value"):
```

Comment:
Do we need the `hasattr(self, key)` check? Or can we do with just the `in` expression?

Reply:
yeah, I tried it without it first, but then it fails on the initialization of the dataclass instance.

Reply:
ah makes sense